tools/docker/ubuntu18.04-devel-gpu.Dockerfile

```diff
@@ -0,0 +1,186 @@
+FROM nvidia/cuda:10.2-cudnn7-devel-ubuntu18.04
+
+LABEL maintainer="GluonNLP Team"
+
+ARG PYTHON=python3
+ARG PIP=pip3
+ARG PYTHON_VERSION=3.7.8
+ARG MX_URL=https://repo.mxnet.io/dist/python/cu102/mxnet_cu102-2.0.0b20200816-py2.py3-none-manylinux2014_x86_64.whl
+
+ARG OPENSSL_VERSION=1.1.1g
+ARG TINI_VERSION=v0.19.0
+
+# Update for tzdata
+ARG DEBIAN_FRONTEND=noninteractive
+ENV TZ=US
+RUN ln -fs /usr/share/zoneinfo/America/New_York /etc/localtime
+
+ENV PYTHONDONTWRITEBYTECODE=1 \
+    PYTHONUNBUFFERED=1 \
+    LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/usr/local/lib" \
+    PYTHONIOENCODING=UTF-8 \
+    LANG=C.UTF-8 \
+    LC_ALL=C.UTF-8
+
+ENV WORKDIR=/workspace
+
+RUN apt-get update \
+ && apt-get install -y --no-install-recommends \
+    software-properties-common \
+    build-essential \
+    ca-certificates \
+    tzdata \
+    curl \
+    emacs \
+    subversion \
+    locales \
+    cmake \
+    git \
+    libopencv-dev \
+    openssh-client \
+    openssh-server \
+    htop \
+    vim \
+    wget \
+    unzip \
+    sudo \
+    python3-dev \
+    python3-pip \
+    python3-setuptools \
+    libxft-dev \
+    zlib1g-dev \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/*
+
+###########################################################################
+# Horovod dependencies
+###########################################################################
+
+# Install Open MPI
+RUN mkdir /tmp/openmpi \
+ && cd /tmp/openmpi \
+ && curl -fSsL -O https://download.open-mpi.org/release/open-mpi/v4.0/openmpi-4.0.1.tar.gz \
+ && tar zxf openmpi-4.0.1.tar.gz \
+ && cd openmpi-4.0.1 \
+ && ./configure --enable-orterun-prefix-by-default \
+ && make -j $(nproc) all \
+ && make install \
+ && ldconfig \
+ && rm -rf /tmp/openmpi
+
+# Create a wrapper for OpenMPI to allow running as root by default
+RUN mv /usr/local/bin/mpirun /usr/local/bin/mpirun.real \
+ && echo '#!/bin/bash' > /usr/local/bin/mpirun \
+ && echo 'mpirun.real --allow-run-as-root "$@"' >> /usr/local/bin/mpirun \
+ && chmod a+x /usr/local/bin/mpirun
+
+RUN echo "hwloc_base_binding_policy = none" >> /usr/local/etc/openmpi-mca-params.conf \
+ && echo "rmaps_base_mapping_policy = slot" >> /usr/local/etc/openmpi-mca-params.conf
+
+ENV LD_LIBRARY_PATH=/usr/local/openmpi/lib:$LD_LIBRARY_PATH
+ENV PATH=/usr/local/openmpi/bin/:/root/.local/bin:$PATH
+
+RUN wget -c https://www.openssl.org/source/openssl-${OPENSSL_VERSION}.tar.gz \
+ && apt-get update \
+ && apt remove -y --purge openssl \
+ && rm -rf /usr/include/openssl \
+ && apt-get install -y \
+    ca-certificates \
+ && tar -xzvf openssl-${OPENSSL_VERSION}.tar.gz \
+ && cd openssl-${OPENSSL_VERSION} \
+ && ./config && make -j $(nproc) && make test \
+ && make install \
+ && ldconfig \
+ && cd .. && rm -rf openssl-*
```

Comment:
Why?

Reply:
Do you mean why should we install openssl?

Reply:
Why compile openssl from source?

Reply:
I guess it's because we may want to have a customized version of openssl. Also, there might be security issues so that we may not install from another resource.  That's the solution adopted in DLC: https://github.com/aws/deep-learning-containers/blob/95e4d9c9cba8b6dffec61637452b4bbd46bb59bd/mxnet/training/docker/1.6.0/py3/cu101/Dockerfile.gpu#L113-L124

Reply:
DLC has a lot of technical debt and is using ubuntu 16.04. For this container,

Reply:
Ubuntu manages the security fixes for you. No need to compile from source. I recommend you remove this part.